frodo.js

```diff
@@ -1,88 +1,97 @@
-/**
```

Comment:
SCRIPT-Well done with the efficient error handling, it covers all the possible scenarios.